### DISCUSSION TITLE
Brigardier Implementation

### DISCUSSION THREAD

**Discussion by BelgianDev** [Minestom/Minestom#2326] [Help] [answered]
Hey, This may be a weird idea. It's kinda against the whole philosophy of the project. 

But our team uses Mojang's Brigardier everywhere so It could be nice if our game software could also use Brigardier for commands, just to make things simple.

But I'm not sure it's possible without forking Minestom and implementing it, or is there a way to add Brigardier without needing to fork Minestom ?

**Comment by DasBabyPixel** [comment, marked as answer]
You can create a command with a string argument which redirects to brigadier, no problem at all and no need to fork minestom

**Reply by BelgianDev** [reply]
Yeah but this won't allow for autocomplete of commands, and it's a feature we still want to use. It also the major reason why we use Brigadier for commands

**Reply by DasBabyPixel** [reply]
Yes it does allow for autocomplete, you just have to redirect everything properly so brigadier can work with it.
![grafik](https://github.com/user-attachments/assets/65049228-b352-41fe-984d-6ee303dcffb9)

**Reply by BelgianDev** [reply]
I'll need to check on that, but porting every argument type to brigadier means also creating adapter classes for all of them.

Did you use Brigadier on that screenshot ?

**Reply by DasBabyPixel** [reply]
Yes, but I made my life easy by using a single string argument that's greedy and with serverside autocomplete.

**Reply by BelgianDev** [reply]
Yeah it's a good idea, but this also means no autocomplete for entities, like `@e[type=!minecraft:player]` and stuff ?

**Reply by DasBabyPixel** [reply]
Autocomplete for entities also works. 
![grafik](https://github.com/user-attachments/assets/962b96ec-da5a-477e-bf7f-3c1f3cb2a00b)
This is not minestom directly, but paper 1.8.8 with brigadier again
But would also work no problem with minestom

**Reply by BelgianDev** [reply]
I see thank!, I'll tinker a bit and find out. At least I know it's possible without needing to fork minestom.

**Comment by Codestech1** [comment]
> Hey, This may be a weird idea. It's kinda against the whole philosophy of the project.
> 
> But our team uses Mojang's Brigardier everywhere so It could be nice if our game software could also use Brigardier for commands, just to make things simple.
> 
> But I'm not sure it's possible without forking Minestom and implementing it, or is there a way to add Brigardier without needing to fork Minestom ?

Good luck with convincing them to move to brigadier. I tried, but they still want to use their command system.

**Reply by BelgianDev** [reply]
I don't really want them to use it to brigadier but it would be nice if we could specify a custom Command Dispatcher, and then pass them a custom implementation of that interface that uses brigadier
